frappe/public/js/frappe/list/base_list.js

```diff
@@ -646,9 +659,25 @@ frappe.ui.FilterArea = class FilterArea {
 		fields.map(df => this.list_view.page.add_field(df));
 	}
 
+	make_custom_filters() {
+		let fields = JSON.parse(this.list_view.list_view_settings.filters);
+
+		fields.forEach(field => {
+			field.onchange = () => this.refresh_list_view();
+		});
+
+		fields.map(df => this.list_view.page.add_field(df));
+	}
+
 	get_standard_filters() {
 		const filters = [];
 		const fields_dict = this.list_view.page.fields_dict;
+		let doc_type = "";
+		for (let key in fields_dict) {
+			if (fields_dict[key].df.doctype)
+				doc_type = fields_dict[key].df.doctype;
+		}
```

Comment:
```suggestion
		for (let key in fields_dict) {
			if (fields_dict[key].df.doctype)
				doc_type = fields_dict[key].df.doctype;
		}
```

the baselist has doctype, use that